compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3596,6 +3607,8 @@ private ParserRuleContext getNextRuleInternal(ParserRuleContext currentCtx, int
                 return ParserRuleContext.TYPE_DESC_IN_PARENTHESIS;
             case SELECT_CLAUSE:
                 return ParserRuleContext.SELECT_KEYWORD;
+            case COLLECT_CLAUSE:
+                return ParserRuleContext.COLLECT_KEYWORD;
```

Comment:
We may need to change `seekMatch()` to match `collect` like this,
```java
hasMatch = BallerinaParser.isKeywordMatch(SyntaxKind.COLLECT_KEYWORD, nextToken);
```

Reply:
Yes :-D